tests/test_solvers/test_harbor_solver.py

```
@@ -1125,32 +1274,138 @@ async def test_flush_patch_timeout_writes_trajectory_no_marker(self, tmp_path, m
+                '        Path(os.environ["NEL_TEST_READY_PATH"]).write_text("ready")\n        time.sleep(30)\n',
                 self._EVENTS_TWO_MESSAGES,
             )
         )
         patched = await self._apply_flush_patch(runner)
+        runner.write_text(f'{patched}\n\nif __name__ == "__main__":\n    main()\n')
+
+        env = os.environ.copy()
+        env["NEL_TEST_TRAJECTORY_PATH"] = str(traj_path)
+        ready_path = tmp_path / "ready"
+        env["NEL_TEST_READY_PATH"] = str(ready_path)
+        proc = subprocess.Popen(
```

Ruff (0.15.21)

[warning] Async functions should not create subprocesses with blocking methods

(ASYNC220)